agent_build_refactored/managed_packages/steps/download_build_dependencies/download_build_dependencies.sh

```diff
@@ -1,9 +1,55 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is meant to be executed by the instance of the 'agent_build_refactored.tools.runner.RunnerStep' class.
+# Every RunnerStep provides common environment variables to its script:
+#   SOURCE_ROOT: Path to the projects root.
+#   STEP_OUTPUT_PATH: Path to the step's output directory.
+#
+# This script downloads source files for all libraries that are required to build Python. Since we build Python and its
+# requirements in a legacy OS (such as Centos 6) to achieve binary compatibility, we also have to build many essential
+# build tools because tools from legacy OS's ate too old.
+#
+# It expects next environment variables:
+#
+#   XZ_VERSION: version of XZ Utils to build. Python requirement. ALso required by some make and configure scripts.
+#   PERL_VERSION: version of Perl to build. Required by some make and configure scripts.
+#   TEXINFO_VERSION: version of texinfo to build. Required by some make and configure scripts.
+#   M4_VERSION: version of M4 to build. Required by some make and configure scripts.
+#   AUTOCONF_VERSION: version of autoconf to build. Required by some make and configure scripts.
+#   LIBTOOL_VERSION: version of libtool to build. Required by some make and configure scripts.
+#   HELP2MAN_VERSION: version of help2man to build. Required by some make and configure scripts.
+#   AUTOMAKE_VERSION: version of automake to build. Required by some make and configure scripts.
+#   LZMA_VERSION: version of lzma to build. Python requirement.
+#   ZLIB_VERSION: version of zlib to build. Python requirement. Provides zlib module.
+#   BZIP_VERSION: version of bzip to build. Python requirement. Provides bz2 module.
+#   UTIL_LINUX_VERSION: version of lzma to build. Python requirement. Provides uuid module.
+#   NCURSES_VERSION: version of ncurses to build. Python requirement. Provides curses module.
+#   LIBEDIT_VERSION: version of libedit to build. Python requirement. Provides non-GPL alternative for readline module.
+#   GDBM_VERSION: version of gdbm to build. Python requirement. Provides dbm module.
+#   LIBFFI_VERSION: version of libffi to build. Python requirement. Provides ctypes module and essential for C bindings.
+#   OPENSSL_VERSION: version of OpenSSL to build. Python requirement. Provides ssl module.
+
 set -e
 
+apt update
 
+DEBIAN_FRONTEND=noninteractive apt install -y git curl gnupg gnupg2 xz-utils
 
 
-XZ_PUBLIC_KEY="-----BEGIN PGP PUBLIC KEY BLOCK-----
+# Import XZ public key
```

Comment:
Thanks for adding key verification. And yeah, as you've implemented it - if we want additional security benefits, we indeed to to bundle the actual keys which are used for signing inside the script / repo.